src/utils.rs

```diff
@@ -68,25 +59,22 @@ pub async fn create_embed_response(
     interaction: &mut ApplicationCommandInteraction,
     embed: CreateEmbed,
 ) -> Result<(), ParrotError> {
-    interaction
+    match interaction
         .create_interaction_response(&http, |response| {
             response
                 .kind(InteractionResponseType::ChannelMessageWithSource)
-                .interaction_response_data(|message| message.add_embed(embed))
+                .interaction_response_data(|message| message.add_embed(embed.clone()))
         })
         .await
-        .map_err(Into::into)
-}
-
-pub async fn create_embed_followup(
-    http: &Arc<Http>,
-    interaction: &mut ApplicationCommandInteraction,
-    embed: CreateEmbed,
-) -> Result<Message, ParrotError> {
-    interaction
-        .create_followup_message(&http, |followup| followup.add_embed(embed))
-        .await
-        .map_err(Into::into)
+    {
+        Ok(val) => Ok(val),
+        Err(err) => match err {
+            SerenityError::Http(..) => edit_embed_response(http, interaction, embed)
+                .await
+                .map(|_| ()),
+            _ => Err(err).map_err(Into::into),
+        },
```

Comment:
Here's a suggestion if you wish to specifically look at the "Interaction has already been acknowledged." error instead of a generic Error::Http. It looks really ugly though, maybe someone with better understanding of box deref can clean it up.

```suggestion
        Err(err) => match err {
            ParrotError::Serenity(Error::Http(ref e)) => match &**e {
                HttpError::UnsuccessfulRequest(req) => match req.error.code {
                    40060 => edit_embed_response(http, interaction, embed)
                        .await
                        .map(|_| ()),
                    _ => Err(err),
                },
                _ => Err(err),
            },
            _ => Err(err),
        },
```

Reply:
That, uh... That is pretty rough, and it doesn't seem to work. At least the compiler doesn't seem to like it. I'm working on making it a little more specific at the moment in a slightly different way. Should be able to test it later tonight.

Reply:
Odd, it compiles and works just fine on my end. Regardless, it _is_ pretty rough, just a suggestion to get you started. 🙂

Reply:
I'm mad, there's not a cleaner way to do this because the damn box match is still experimental. Oh well. Accepting this with a little cleaning up of my stuff.